### ISSUE TITLE
Customizing the CSS identifiers (scope, hash etc)

### ISSUE THREAD

**Issue by katspaugh** [vercel/next.js#9888]
# Feature request

## Problem
Our Selenium tests rely on a specific naming of class names in the HTML. E.g. `class="Header__link"`, where `Header` is the name of a component, and `link` is a class in the component's CSS module).

With the current experimental CSS support in Next.js, there's no apparent way to customize the outputted CSS identifiers.

With `next-css`, we used to simply override the Webpack config with our own css-loader options `localIdentName` and `getLocalIdent`).

## Proposed solution

Please allow to override the Webpack config like it was the case with `next-css`.
Thanks!

Related to #8626.

**Comment by Timer** [comment]
How do you guarantee uniqueness throughout your whole application?

**Comment by katspaugh** [comment]
In our case the file name is always `styles.module.css`. It’s the parent directory that corresponds to the component name.

The component name (or rather its path) guarantees uniqueness. Another example of our former identifiers: `signup_FacebookButton__icon`. The CSS is in `components/signup/FacebookButton/styles.module.css`.

**Comment by Timer** [comment]
To set expectations, we're probably not going to ship with this ability by default.

We can reconsider this when it's released stable if there's significant demand. IMO, there's better workarounds here.

For example, if you are writing Unit Test using Jest, you should be [eliminating CSS Module exports](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/config/jest/cssTransform.js) **or** using [`identity-obj-proxy`](https://github.com/keyz/identity-obj-proxy) which would skip hashing.

If you are writing Integration Tests, you should be using [`getComputedStyle()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle) — not class names.

**Comment by katspaugh** [comment]
Thanks for the reply. I think your reasoning and strategy makes total sense. Also less configuration for the user.

I guess we'll have to use `next-css` until it stops working or and rewrite all the tests in the meanwhile. However, if there is demand after all, I suggest people interested in this feature to upvote the issue.

Cheers 🙂

**Comment by Timer** [comment]
Your use case would probably be served well by [`identity-obj-proxy`](https://github.com/keyz/identity-obj-proxy) — but yes, you can keep using `@zeit/next-css`; we're not going to maintain it but we're not going to intentionally break or block it either. 😄

**Comment by morganfeeney** [comment]
> tl;dr: To set expectations, we're probably not going to ship with this ability by default. Read on for testing alternatives.

@Timer when you say not ship with the ability to do this, are you referring to what has been requested in this issue, or the ability to turn on the experimental CSS features?

I'm using BEM and next-css and well... this is a big change, I'm using a well documented system for creating components (BEM), and I'm concerned that the new CSS features are going to mean that we'll end up having to re-invent the wheel by creating something that works with a CSS module based approach.
